empowering_api/empowering_api.py

```diff
@@ -56,3 +56,113 @@ def __init__(self, pool, cursor):
+                                                   ('district_heating', 'District heating'),
+                                                   ('biomass', 'Biomass'),
+                                                   ('other', 'Other')],
+                                                  'Building heating source'),
+        'buildingHeatingSourceDhw': fields.selection([('electricity', 'Electricity'),
+                                                      ('gas', 'Gas'),
+                                                      ('gasoil', 'Gasoil'),
+                                                      ('district_heating', 'District heating'),
+                                                      ('biomass', 'Biomassa'),
+                                                      ('other', 'Other')],
+                                                     'Building heating source dhw'),
+        'buildingSolarSystem': fields.selection([('PV'),
+                                                 ('solar_thermal_heating', 'Solar thermal heating'),
+                                                 ('solar_thermal_DHW', 'Solar thermal DHW'),
+                                                 ('other', 'Other'),
+                                                 ('not_installed', 'Not installed')],
+                                                'Building solar system')
+    }
+
+EmpoweringCupsBuilding()
+
+
+class EmpoweringModcontractualProfile(osv.osv):
+    _name = 'empowering.modcontractual.profile'
+
+    _columns = {
+        'modcontractual_id': fields.many2one('giscedata.polissa.modcontractual', 'Contract Modification', required=True,
+                                             readonly=True, select=1),
+        'totalPersonsNumber': fields.integer('Total persons number'),
+        'minorPersonsNumber': fields.integer('Minor persons number'),
+        'workingAgePersonsNumber': fields.integer('Working age persons number'),
+        'retiredAgePersonsNumber': fields.integer('Retired age persons number'),
+        'malePersonsNumber': fields.integer('Male persons number'),
+        'femalePersonsNumber': fields.integer('Female persons number'),
+        'eduLevel_prim': fields.integer('Primary education level persons number'),
+        'eduLevel_sec': fields.integer('Secondary education level persons number'),
+        'eduLevel_uni': fields.integer('University education level persons number'),
+        'eduLevel_noStudies': fields.integer('No studies education level persons number')
+    }
+
+EmpoweringModcontractualProfile()
+
+
+class EmpoweringModcontractualService(osv.osv):
+    _name = 'empowering.modcontractual.service'
+
+    _columns = {
+        'modcontractual_id': fields.many2one('giscedata.polissa.modcontractual', 'Contract modification', required=True,
+                                             readonly=True, select=1),
+        'OT101': fields.char('OT101', size=100),
```

Comment:
L'he definit com a char per a la definició de l'API

*customisedGroupingCriteria
In the case the utility wants to impose a personalised grouping criterion (e.g. criteria_1), here the value can be inserted. In this way, when indicated this criterion to be used, the grouping can be done over all contracts having the same value (CLASS 1).*